### PR TITLE
makes previous results available to rollback function

### PR DESCRIPTION
**Pull request by jwebber18**

### CODE REVIEW

**Review by thebriz24 — COMMENTED** [review]
@jwebber18 So the point of the `parser` was to declare what you wanted from the return value so that it'd be available for the rollback. I was modeling it after a `with` statement where error handling doesn't have access to any of the step results other than what is pattern matched from the current step. It's done to ensure reliability of the handlers. I'm worried that simply having access to the `previous` field will increase the possibility of errors. Thoughts?

**Comment by jwebber18** [comment]
@thebriz24 the main advantage I see of having access to the `previous` field is for update rollbacks. For example, we frequently do something like:

operation 1: fetch struct
operation 2: update struct

To rollback an update, we need to know the original state, which we could get from operation 1. But we currently only have access to the updated version. I suppose we could then have the parser return the old version, but we might need access to the updated version in later operations in the Manifest or as part of the return.

**Comment by thebriz24** [comment]
@jwebber18 I'm realizing as I'm formulating my response that I believe I am using the wrong data structure to really do this properly. So I am going to spin up an issue that I will use for v1.0. I'll link it here once I've finished thinking through how it needs to happen. I'll approve this so you're not held back while I work on it. I'm just going to request a few changes for compatibility.

File: lib/manifest.ex [hunk 1]
```diff
@@ -271,8 +271,8 @@ defmodule Manifest do
 
   defp rollback([], acc), do: {:ok, acc}
 
-  defp rollback([{operation, {rollback, identifier}} | rest], acc) do
-    case rollback.(identifier) do
+  defp rollback([{operation, {rollback, identifier, previous}} | rest], acc) do
+    case rollback.(identifier, previous) do
```
Comment (thebriz24) [inline lib/manifest.ex]: Can you make this work with both an arity of 1 and 2? Most likely it'll have to be a try block.

**Review by thebriz24 — CHANGES_REQUESTED** [review]
Also bump the patch version please.

**Review by thebriz24 — APPROVED** [review]

**Pull request merged**